File: LeetCode/3301-3400/Problem3304_FindtheKthCharacterinStringGameI/FindtheKthCharacterinStringGameI.cpp
class Solution {
  public:
      char kthCharacter(int k) {
          std::string word = "a";
          
          while (word.length() < k) {
              std::string nextPart;
              for (char c : word) {
                  char nextChar = (c - 'a' + 1) % 26 + 'a';
                  nextPart += nextChar;
              }
              word += nextPart;
          }
  
          return word[k - 1];
      }
  };
  

File: LeetCode/3301-3400/Problem3304_FindtheKthCharacterinStringGameI/FindtheKthCharacterinStringGameI.rs
impl Solution {
    pub fn kth_character(k: i32) -> char {
        let mut word = String::from("a");

        while word.len() < k as usize {
            let next_part: String = word
                .chars()
                .map(|c| ((c as u8 - b'a' + 1) % 26 + b'a') as char)
                .collect();
            word.push_str(&next_part);
        }

        word.chars().nth((k - 1) as usize).unwrap()
    }
}
